plugin/core/types.py

```diff
@@ -719,6 +713,17 @@ def is_disabled_capability(self, capability_path: str) -> bool:
                     return True
         return False
 
+    def filter_disabled_capabilities(self, capability_path: str, options: Dict[str, Any]) -> bool:
+        if self.is_disabled_capability(capability_path):
+            return True
+        to_be_removed = []  # type: List[str]
+        for k in options.keys():
+            if self.is_disabled_capability("{}.{}".format(capability_path, k)):
+                to_be_removed.append(k)
+        for k in to_be_removed:
+            options.pop(k)
+        return False
+
```

Comment:
This is a rather confusing method:
 - It's not that obvious what that return value means
 - I'm not sure if the name really describes well what it does.
 - Mutating the passed-in `options` doesn't seem like a great practice

How about changing it like so:
 - return a new object with only enabled capabilities
 - don't check `is_disabled_capability` here, move it outside of this method
 - call it `filter_out_disabled_capabilities`? I feel like "filter" can mean either "return only disabled capabilities" or "return only enabled capabilities"

Reply:
fixed in 08c6c80